testing/hardlinktest.py

```diff
@@ -156,6 +156,121 @@ def testInnerRestore(self):
         assert hlout2.getinode() == hlout3.getinode()
         assert hlout1.getinode() != hlout2.getinode()
 
+    def testHardlinkBug1(self):
```

Comment:
Please give  a more descriptive function name and a short description of the bug you're trying to reproduce and how. In X years, nobody will care about the initial bug report but the test might still be relevant.

I would also have expected a test that:
1. there is no warning about missing hash when checking the repo.
2. no discrepancy in number of links in the metadata

And I would suggest to backup with an explicit `current_time` and restore with the same time to make it clearer what you're trying to do. Other tests use 10000, 20000, etc... You can then also remove the 1s wait time.

Reply:
Thanks.  I plan to work on this today.